### PR TITLE
Add Nametags and fix interdm teleports

### PR DESCRIPTION
**Pull request by Raycoms**
This should be fixing inter dim teleportation.

There are some issues on a server atm. I read that end tp has some extra cases so we have to consider those extra .
I tested all of this, it seems to be working fine.

I added also a way to change the citizen names via name tags.

Review please

### CODE REVIEW

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 32 issues
* ![MINOR][MINOR] 32 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/caee7df0b0e294700b19ba199f086f6a30a07c9a/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/caee7df0b0e294700b19ba199f086f6a30a07c9a/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render#](https://github.com/Minecolonies/minecolonies/blob/caee7df0b0e294700b19ba199f086f6a30a07c9a/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/caee7df0b0e294700b19ba199f086f6a30a07c9a/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/caee7df0b0e294700b19ba199f086f6a30a07c9a/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/caee7df0b0e294700b19ba199f086f6a30a07c9a/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [requestsystem#](https://github.com/Minecolonies/minecolonies/blob/caee7df0b0e294700b19ba199f086f6a30a07c9a/src/main/java/com/minecolonies/coremod/colony/requestsystem#): Add a 'package-info.java' file to document the 'requestsystem' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [init#](https://github.com/Minecolonies/minecolonies/blob/caee7df0b0e294700b19ba199f086f6a30a07c9a/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/caee7df0b0e294700b19ba199f086f6a30a07c9a/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [requests#](https://github.com/Minecolonies/minecolonies/blob/caee7df0b0e294700b19ba199f086f6a30a07c9a/src/main/java/com/minecolonies/coremod/colony/requestsystem/requests#): Add a 'package-info.java' file to document the 'requests' package [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1228)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Asherslab** [comment]
Loooooks good to me!! :shipit:

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1695/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Review by marchermans — APPROVED** [review]
Code wise this looks good. No idea if it works. Never tried custom dimension teleporting.

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 69 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 30 major
* ![MINOR][MINOR] 32 minor
* ![INFO][INFO] 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L190](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L190): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L191](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L191): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L192](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L192): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L193](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L193): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L194](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L194): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [CitizenData.java#L26](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L26): class "CitizenData" has 45 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [CitizenData.java#L237](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L237): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [CitizenData.java#L518](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L518): Use the "equals" method if value comparison was intended. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1698)
1. ![MAJOR][MAJOR] [CitizenData.java#L523](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L523): Use the "equals" method if value comparison was intended. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1698)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 1]
```diff
@@ -1271,6 +1274,35 @@ private void cleanupChatMessages()
...
     }
 
+    @Override
+    public void setCustomNameTag(final String name)
+    {
+        if(citizenData != null && name != null && Configurations.gameplay.allowGlobalNameChanges >= 0)
+        {
+            if(Configurations.gameplay.allowGlobalNameChanges == 0 &&
+                    Arrays.stream(Configurations.gameplay.specialPermGroup).anyMatch(owner -> owner.equals(colony.getPermissions().getOwnerName())))
+            {
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: MAJOR Either remove or fill this block of code. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS00108)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 70 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 31 major
* ![MINOR][MINOR] 32 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L190](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L190): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L191](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L191): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L192](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L192): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L193](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L193): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L194](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L194): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [CitizenData.java#L26](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L26): class "CitizenData" has 46 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [CitizenData.java#L237](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L237): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [CitizenData.java#L518](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L518): Use the "equals" method if value comparison was intended. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1698)
1. ![MAJOR][MAJOR] [CitizenData.java#L523](https://github.com/Minecolonies/minecolonies/blob/9475b64505f2bd5aadd9e08d39948bcce0cb354f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L523): Use the "equals" method if value comparison was intended. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1698)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 2]
```diff
@@ -1271,6 +1274,44 @@ private void cleanupChatMessages()
...
+                            return;
+                        }
+                    }
+                    this.citizenData.setName(name);
+                    this.citizenData.markDirty();
+                    super.setCustomNameTag(name);
+                }
+            }
+            else
+            {
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java]: MAJOR Either remove or fill this block of code. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS00108)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 3]
```diff
@@ -1271,6 +1274,44 @@ private void cleanupChatMessages()
         }
     }
 
+    @Override
+    public void setCustomNameTag(final String name)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:1280]: MAJOR Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS3776)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 71 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 32 major
* ![MINOR][MINOR] 32 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L190](https://github.com/Minecolonies/minecolonies/blob/5ecf4ccae99432895a02ad93e9a43a8b2f2b5d3d/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L190): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L191](https://github.com/Minecolonies/minecolonies/blob/5ecf4ccae99432895a02ad93e9a43a8b2f2b5d3d/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L191): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L192](https://github.com/Minecolonies/minecolonies/blob/5ecf4ccae99432895a02ad93e9a43a8b2f2b5d3d/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L192): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L193](https://github.com/Minecolonies/minecolonies/blob/5ecf4ccae99432895a02ad93e9a43a8b2f2b5d3d/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L193): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L194](https://github.com/Minecolonies/minecolonies/blob/5ecf4ccae99432895a02ad93e9a43a8b2f2b5d3d/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L194): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/5ecf4ccae99432895a02ad93e9a43a8b2f2b5d3d/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [CitizenData.java#L26](https://github.com/Minecolonies/minecolonies/blob/5ecf4ccae99432895a02ad93e9a43a8b2f2b5d3d/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L26): class "CitizenData" has 46 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [CitizenData.java#L237](https://github.com/Minecolonies/minecolonies/blob/5ecf4ccae99432895a02ad93e9a43a8b2f2b5d3d/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L237): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [CitizenData.java#L518](https://github.com/Minecolonies/minecolonies/blob/5ecf4ccae99432895a02ad93e9a43a8b2f2b5d3d/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L518): Use the "equals" method if value comparison was intended. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1698)
1. ![MAJOR][MAJOR] [CitizenData.java#L523](https://github.com/Minecolonies/minecolonies/blob/5ecf4ccae99432895a02ad93e9a43a8b2f2b5d3d/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L523): Use the "equals" method if value comparison was intended. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1698)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 70 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 31 major
* ![MINOR][MINOR] 32 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L190](https://github.com/Minecolonies/minecolonies/blob/e868644b29c81c3c4c263b54ae2f1d6275d60bd2/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L190): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L191](https://github.com/Minecolonies/minecolonies/blob/e868644b29c81c3c4c263b54ae2f1d6275d60bd2/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L191): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L192](https://github.com/Minecolonies/minecolonies/blob/e868644b29c81c3c4c263b54ae2f1d6275d60bd2/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L192): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L193](https://github.com/Minecolonies/minecolonies/blob/e868644b29c81c3c4c263b54ae2f1d6275d60bd2/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L193): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L194](https://github.com/Minecolonies/minecolonies/blob/e868644b29c81c3c4c263b54ae2f1d6275d60bd2/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L194): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L195](https://github.com/Minecolonies/minecolonies/blob/e868644b29c81c3c4c263b54ae2f1d6275d60bd2/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L195): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [CitizenData.java#L26](https://github.com/Minecolonies/minecolonies/blob/e868644b29c81c3c4c263b54ae2f1d6275d60bd2/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L26): class "CitizenData" has 46 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [CitizenData.java#L237](https://github.com/Minecolonies/minecolonies/blob/e868644b29c81c3c4c263b54ae2f1d6275d60bd2/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L237): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [CitizenData.java#L518](https://github.com/Minecolonies/minecolonies/blob/e868644b29c81c3c4c263b54ae2f1d6275d60bd2/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L518): Use the "equals" method if value comparison was intended. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1698)
1. ![MAJOR][MAJOR] [CitizenData.java#L523](https://github.com/Minecolonies/minecolonies/blob/e868644b29c81c3c4c263b54ae2f1d6275d60bd2/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L523): Use the "equals" method if value comparison was intended. [rule](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1698)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**